core/src/main/java/com/linecorp/armeria/common/HttpRequestDuplicator.java

```diff
@@ -16,128 +16,69 @@
 
 package com.linecorp.armeria.common;
 
-import static java.util.Objects.requireNonNull;
+import org.reactivestreams.Subscriber;
 
-import javax.annotation.Nullable;
-
-import com.google.common.base.MoreObjects;
-
-import com.linecorp.armeria.common.stream.AbstractStreamMessageDuplicator;
-import com.linecorp.armeria.common.stream.StreamMessage;
-import com.linecorp.armeria.common.stream.StreamMessageWrapper;
-
-import io.netty.util.concurrent.EventExecutor;
+import com.linecorp.armeria.common.stream.StreamMessageDuplicator;
+import com.linecorp.armeria.common.stream.SubscriptionOption;
 
 /**
- * Allows subscribing to an {@link HttpRequest} multiple times by duplicating the stream.
+ * A duplicator that duplicates a {@link HttpRequest} into one or more {@link HttpRequest}s,
+ * which publish the same elements.
  *
  * <pre>{@code
- * final HttpRequest originalReq = ...
- * final HttpRequestDuplicator reqDuplicator = new HttpRequestDuplicator(originalReq);
- *
- * final HttpRequest dupReq1 = reqDuplicator.duplicateStream();
- * final HttpRequest dupReq2 = reqDuplicator.duplicateStream(true); // the last stream
+ * HttpRequest<String> httpRequest = ...
```

Comment:
`c.l.a.c.HttpRequest` doesn't seem to have type parameter. 😀

Reply:
Oops. 😅